tests/lib/rook/base.py

```diff
@@ -40,6 +44,52 @@ def destroy(self, skip=True):
         # TODO(jhesketh): Uninstall rook
         pass
 
+    def execute_in_ceph_toolbox(self, command):
+        if not self.toolbox_pod:
+            self.toolbox_pod = self.kubernetes.get_pod_by_app_label(
+                "rook-ceph-tools")
+
+        return self.kubernetes.execute_in_pod(command, self.toolbox_pod)
+
+    @abstractmethod
+    def preinstall(self):
```

Comment:
not implemented in the `upstream.py` implementation

Reply:
I removed it for now. Will fix upstream and add this in another PR.